keel-core/src/main/kotlin/com/netflix/spinnaker/keel/logging/TracingSupport.kt

```diff
@@ -0,0 +1,42 @@
+package com.netflix.spinnaker.keel.logging
+
+import com.netflix.spinnaker.keel.api.Exportable
+import com.netflix.spinnaker.keel.api.Resource
+import com.netflix.spinnaker.keel.api.ResourceId
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.api.id
+import kotlinx.coroutines.CoroutineScope
+import org.slf4j.MDC
+import kotlinx.coroutines.slf4j.MDCContext
+import kotlinx.coroutines.withContext
+
+/**
+ * Support for tracing resources in log statements via MDC in coroutines.
+ */
+class TracingSupport {
+  companion object {
+    const val X_SPINNAKER_RESOURCE_ID = "X-SPINNAKER-RESOURCE-ID"
```

Comment:
Not sure if `kork` is used in `keel`... if it is: https://github.com/spinnaker/kork/blob/master/kork-security/src/main/java/com/netflix/spinnaker/security/AuthenticatedRequest.java#L66

Reply:
It is! Technically, this is not a header though, and not necessarily restricted to web request flows as exemplified by the scheduled resource checks flow instrumented in this PR. Would you still recommend it?

Reply:
i guess if it's not a header or mdc probably leave as is.

Reply:
We probably do want to propagate it as a request header, keel can make plenty of downstream calls to other spinnaker µservices in the context of evaluating a resource.

Reply:
Sure, we can look into that, but in a separate PR, if you don't mind. 😊

Reply:
👍